service/src/main/java/greencity/service/HabitFactServiceImpl.java

```diff
@@ -121,18 +124,20 @@ public HabitFactVO save(HabitFactPostDto fact) {
     @Override
     public HabitFactVO update(HabitFactUpdateDto factDto, Long id) {
         HabitFact habitFact = habitFactRepo.findById(id)
-            .orElseThrow(() -> new NotUpdatedException(ErrorMessage.ADVICE_NOT_FOUND_BY_ID + id));
-        List<HabitFactTranslation> habitFactTranslations = modelMapper.map(factDto.getTranslations(),
-            new TypeToken<List<HabitFactTranslation>>() {
-            }.getType());
-        habitFactTranslationRepo.deleteAllByHabitFact(habitFact);
-        habitFact.setTranslations(habitFactTranslations);
-        habitFactTranslations.forEach(habitFactTranslation -> {
-            habitFactTranslation.setHabitFact(habitFact);
-            habitFactTranslation.setFactOfDayStatus(factDto.getTranslations().get(0).getFactOfDayStatus());
-        });
-        habitFactRepo.save(habitFact);
-        return modelMapper.map(habitFact, HabitFactVO.class);
+            .orElseThrow(() -> new NotUpdatedException(ErrorMessage.HABIT_FACT_NOT_FOUND_BY_ID + id));
+        Habit habit = habitRepo.findById(factDto.getHabit().getId())
+            .orElseThrow(() -> new NotUpdatedException(ErrorMessage.HABIT_NOT_FOUND_BY_ID + id));
+        habitFact.setHabit(habit);
+        habitFact.getTranslations()
+            .forEach(habitFactTranslation -> {
+                habitFactTranslation.setContent(factDto.getTranslations().stream()
+                    .filter(newTranslation -> newTranslation.getLanguage().getCode()
+                        .equals(habitFactTranslation.getLanguage().getCode()))
+                    .findFirst().get()
+                    .getContent());
+                habitFactTranslation.setFactOfDayStatus(factDto.getTranslations().get(0).getFactOfDayStatus());
+            });
```

Comment:
Move it to private method with proper name

Reply:
done